evalml/tests/data_checks_tests/test_id_columns_data_check.py

```diff
@@ -168,6 +141,171 @@ def test_id_columns_strings():
+            data_check_name=id_data_check_name,
+            message_code=DataCheckMessageCode.HAS_ID_COLUMN,
+            details={"columns": [0, 1]},
+            action_options=[
+                DataCheckActionOption(
+                    DataCheckActionCode.DROP_COL,
+                    data_check_name=id_data_check_name,
+                    metadata={"columns": [0, 1]},
+                ),
+            ],
+        ).to_dict(),
+    ]
+
+    #  test 2D list
+    assert id_cols_check.validate([[0, 1], [1, 2], [2, 3], [3, 4], [4, 5]]) == [
+        DataCheckWarning(
+            message="Columns '0', '1' are 80.0% or more likely to be an ID column",
+            data_check_name=id_data_check_name,
+            message_code=DataCheckMessageCode.HAS_ID_COLUMN,
+            details={"columns": [0, 1]},
+            action_options=[
+                DataCheckActionOption(
+                    DataCheckActionCode.DROP_COL,
+                    data_check_name=id_data_check_name,
+                    metadata={"columns": [0, 1]},
+                ),
+            ],
+        ).to_dict(),
+    ]
+
+    # test np.array
+    assert id_cols_check.validate(
+        np.array([[0, 1], [1, 2], [2, 3], [3, 4], [4, 5]]),
+    ) == [
+        DataCheckWarning(
+            message="Columns '0', '1' are 80.0% or more likely to be an ID column",
+            data_check_name=id_data_check_name,
+            message_code=DataCheckMessageCode.HAS_ID_COLUMN,
+            details={"columns": [0, 1]},
+            action_options=[
+                DataCheckActionOption(
+                    DataCheckActionCode.DROP_COL,
+                    data_check_name=id_data_check_name,
+                    metadata={"columns": [0, 1]},
+                ),
+            ],
+        ).to_dict(),
+    ]
+
+
+def test_identified_first_col_primary_key():
+    X_dict = {
+        "col_1_id": [0, 1, 2, 3],
+        "col_2": [2, 3, 4, 5],
+        "col_3_id": [1, 1, 2, 3],
+        "col_5": [0, 0, 1, 2],
+        "col_6": [0.1, 0.2, 0.3, 0.4],
+    }
+
+    X = pd.DataFrame.from_dict(X_dict)
+    id_cols_check = IDColumnsDataCheck(id_threshold=0.95)
+    assert id_cols_check.validate(X) == [
+        DataCheckWarning(
+            message="The first column 'col_1_id' has a high likelihood of being the primary key. Columns 'col_2', 'col_3_id' are 95.0% or more likely to be an ID column",
+            data_check_name=id_data_check_name,
+            message_code=DataCheckMessageCode.HAS_ID_FIRST_COLUMN,
+            details={"primary_key": "col_1_id", "columns": ["col_2", "col_3_id"]},
+            action_options=[
+                DataCheckActionOption(
+                    DataCheckActionCode.SET_FIRST_COL_ID,
+                    data_check_name=id_data_check_name,
+                    metadata={
+                        "primary_key": "col_1_id",
+                        "columns": ["col_2", "col_3_id"],
```

Comment:
Curious what the others think about this. I think it'd be ideal if the `SET_FIRST_COL_ID` data check action is only associated with the column that we want to set as the primary key, and `DROP_COL` data check action to be associated with the other columns, rather than associating those columns with the `SET_FIRST_COL_ID` action. I don't think this is blocking, but might be something to file for future fix if we don't get to it now.